### ISSUE TITLE
Facilitar a inserção de novos cadastros.

### ISSUE THREAD

**Issue by cauequeiroz** [pokemongovet/pokemongo.vet.br#23]
Hey folks!

Nesse começo de projeto não temos nenhuma estrutura no backend automatizando as coisas, então todos os cadastros recebidos serão colocados na mão mesmo.  Podemos criar algo para facilitar e organizar esse processo:

- Criar um JSON com os cadastros feitos.
- Adaptar a pagina de listagem para consumir esse JSON e montar tudo dinamicamente.

Assim, quando for adicionar/remover/editar um cadastro, não precisa fazer direto no html, basta alterar o JSON!

**Comment by rndnunes** [comment]
Hey Galera, sou amigo do Cauê ele me apresentou o projeto e achei super interessante. 
Gostaria de contribuir de alguma forma, então vou pegar essa feature para fazer!

**Comment by fabricionaweb** [comment]
Tive essa mesma ideia durante o dia, concordo plenamente. Quero acompanhar para contribuir.

**Comment by fabricionaweb** [comment]
Acredito que essa issue precisa vir após resolvermos #21 pois o formato do JSON depende de como vai ficar a listagem

**Comment by cauequeiroz** [comment]
@fabricionaweb Sim! Eu to no aguardo para ver se comentam algo la nessa issue, porque o pull request com essa alteração ja ta até aberto, é só questão de aprovarem ou nao. Ai esse JSON é algo bem rapido e simples de fazer...

**Comment by lapa182** [comment]
@cauequeiroz Vi a postagem lá no Front End Brasil e resolvi dar uma passada. Tava lendo teu issue aqui no Github, por que não coloca a plataforma dentro do (Assemble)[http://assemble.io]? Basicamente em vez de html faríamos tudo com Handlebars e os cadastros estariam dentro do JSON definido. Se tu quiser eu posso montar um sample project com base nas atuais informações.

**Comment by rcgivisiez** [comment]
Hmm... @lapa182 acho que até podemos ver isso, mas o bacana seria talvez aproveitar pra algo que vá ser usado por um projeto real. Acho que a gente pode procurar algo que vá ser usado em produção.

**Comment by saraiva1989** [comment]
Como está essa issue, vi que a tela abaixo ainda está fixa no HTML e não vem de uma estrutura json.
![image](https://user-images.githubusercontent.com/40599423/138111686-a423a5c8-f924-494f-b631-3862811f330e.png)

Pretendem fazer, se sim posso ajudar.

**Comment by ChristianHerber** [comment]
Olá @saraiva1989, antes de ver esta issue comecei a mexer com algo parecido. O que fiz a princípio foi separar os PokeCenters por estado, cada estado é um arquivo Javascript module que contém um array de objetos e uma chamada para uma função index que lista os items nas divs correspondente ao estado.

![pokeCentersModule](https://user-images.githubusercontent.com/8668150/144416371-d2d17141-e7c0-435b-97b0-73af90635439.jpg)

Aguardo sua reposta sobre esta proposta para podermos dar continuidade.

**Comment by saraiva1989** [comment]
olá @ChristianHerber primeiramente obrigado pelas contribuições e seja bem vindo ao projeto.

em relação a sua implementação nem precisaria se um module javascript, poderia ser um arquivo ,json que é referenciado no HTML (também funcionaria).

Mas de qualquer forma, esse tipo de decisão teria que falar com o @fititnt.

Ele tem um pouco de receio de sair do html e deixar o projeto complexo devido a finalidade do mesmo.

Pelas conversas que tive com ele, entendo que esse projeto tem duas finalidades com o mesmo peso.

1 - A adoção de pet.
2 - inclusão de novos programadores em projetos open source.

referente a essa segunda opção, como o foco é a inclusão de pessoas de fato iniciantes (não que experientes não possa contribuir), mas se deixar complexo, essas pessoas podem ter receio de ajudar.

Claro que criar um modulo, ou um arquivo .js, ler ele via javascript e renderizar via DOM não é complexo para quem tem um pouquinho de experiencia. Mas para quem está começando pode ser.

Não estou dizendo que não será implementado, estou apenas passando a visão o @fititnt me mostrou em conversas. 

Sendo assim vou deixar com ele para avaliar sobre sua proposta de melhoria.

Espero que isso não desmotive e que você entenda o porque ele está seguindo esse caminho.

Caso queira contribuir em algum outro projeto que seja um pouco mais complexo, ai podemos conversar. Me add no https://www.linkedin.com/in/saraiva1989/  e me chame se quiser. Eu demoro um pouco para responder mas sempre respondo rsrs.
